iowrite.cpp

```diff
@@ -978,9 +981,56 @@ bool writeVelocitySpace(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpi
+            cellX = mpiGrid[cells[i]]->parameters[CellParams::XCRD];
+            cellY = mpiGrid[cells[i]]->parameters[CellParams::YCRD];
+            cellZ = mpiGrid[cells[i]]->parameters[CellParams::ZCRD];
+            DX = mpiGrid[cells[i]]->parameters[CellParams::DX];
+            DY = mpiGrid[cells[i]]->parameters[CellParams::DY];
+            DZ = mpiGrid[cells[i]]->parameters[CellParams::DZ];
+            // First corner
+            minRCornerSquare = cellX * cellX + cellY * cellY + cellZ * cellZ;
+            maxRCornerSquare = minRCornerSquare;
+            // Second corner
+            rCornerSquare = (cellX + DX) * (cellX + DX) + cellY * cellY + cellZ * cellZ;
+            if (rCornerSquare > maxRCornerSquare) {maxRCornerSquare = rCornerSquare;}
+            if (rCornerSquare < minRCornerSquare) {minRCornerSquare = rCornerSquare;}
+            // Third corner
+            rCornerSquare = cellX * cellX + (cellY + DY) * (cellY + DY) + cellZ * cellZ;
+            if (rCornerSquare > maxRCornerSquare) {maxRCornerSquare = rCornerSquare;}
+            if (rCornerSquare < minRCornerSquare) {minRCornerSquare = rCornerSquare;}
+            // Fourth corner
+            rCornerSquare = (cellX + DX) * (cellX + DX) + (cellY + DY) * (cellY + DY) + cellZ * cellZ;
+            if (rCornerSquare > maxRCornerSquare) {maxRCornerSquare = rCornerSquare;}
+            if (rCornerSquare < minRCornerSquare) {minRCornerSquare = rCornerSquare;}
+            // Fifth corner
+            rCornerSquare = cellX * cellX + cellY * cellY + (cellZ + DZ) * (cellZ + DZ);
+            if (rCornerSquare > maxRCornerSquare) {maxRCornerSquare = rCornerSquare;}
+            if (rCornerSquare < minRCornerSquare) {minRCornerSquare = rCornerSquare;}
+            // Sixth corner
+            rCornerSquare = (cellX + DX) * (cellX + DX) + cellY * cellY + (cellZ + DZ) * (cellZ + DZ);
+            if (rCornerSquare > maxRCornerSquare) {maxRCornerSquare = rCornerSquare;}
+            if (rCornerSquare < minRCornerSquare) {minRCornerSquare = rCornerSquare;}
+            // Seventh corner
+            rCornerSquare = cellX * cellX + (cellY + DY) * (cellY + DY) + (cellZ + DZ) * (cellZ + DZ);
+            if (rCornerSquare > maxRCornerSquare) {maxRCornerSquare = rCornerSquare;}
+            if (rCornerSquare < minRCornerSquare) {minRCornerSquare = rCornerSquare;}
+            // Eighth corner
+            rCornerSquare = (cellX + DX) * (cellX + DX) + (cellY + DY) * (cellY + DY) + (cellZ + DZ) * (cellZ + DZ);
+            if (rCornerSquare > maxRCornerSquare) {maxRCornerSquare = rCornerSquare;}
+            if (rCornerSquare < minRCornerSquare) {minRCornerSquare = rCornerSquare;}
+            if (minRCornerSquare <= shellRadiusSquare && maxRCornerSquare > shellRadiusSquare) {
+               velSpaceCells.push_back(cells[i]);
+               mpiGrid[cells[i]]->parameters[CellParams::ISCELLSAVINGF] = 1.0;
+            }
+         }
```

Comment:
Lots of ifs. How about:
dx_rm = cellX < 0 ? DX : 0; 
dx_rp = cellX < 0 ? 0 : DX; 
dy_rm = cellY < 0 ? DY : 0; 
dy_rp = cellY < 0 ? 0 : DY; 
dz_rm = cellZ < 0 ? DZ : 0; 
dz_rp = cellZ < 0 ? 0 : DZ;
rsquare_minus = (cellX+dx_rm)*(cellX+dx_rm) + (cellY+dy_rm)*(cellY+dy_rm) + (cellZ+dz_rm)*(cellZ+dz_rm);
rsquare_plus = (cellX+dx_rp)*(cellX+dx_rp) + (cellY+dy_rp)*(cellY+dy_rp) + (cellZ+dz_rp)*(cellZ+dz_rp);
if (rsquare_minus <= shellRadiusSquare && rsquare_plus > shellRadiusSquare) {
               velSpaceCells.push_back(cells[i]);
               mpiGrid[cells[i]]->parameters[CellParams::ISCELLSAVINGF] = 1.0;
            }

Reply:
Implemented as suggested.